src/modules/ejector/ejector.py

```diff
@@ -240,10 +256,32 @@ def _get_predicted_withdrawable_epoch(
 
         churn_limit = self._get_churn_limit(blockstamp)
 
-        epochs_required_to_exit_validators = (validators_to_eject_count + latest_to_exit_validators_count) // churn_limit
+        epochs_required_to_exit_validators = (len(validators_to_eject) + 1 + latest_to_exit_validators_count) // churn_limit
 
         return EpochNumber(max_exit_epoch_number + epochs_required_to_exit_validators + MIN_VALIDATOR_WITHDRAWABILITY_DELAY)
 
+    def _get_predicted_withdrawable_epoch_post_electra(
+        self,
+        blockstamp: ReferenceBlockStamp,
+        validators_to_eject: list[Validator],
+    ) -> EpochNumber:
+        per_epoch_churn = compute_exit_balance_churn_limit(self._get_total_active_balance(blockstamp))
+        activation_exit_epoch = compute_activation_exit_epoch(blockstamp.ref_epoch)
+        state_view = self.w3.cc.get_state_view(blockstamp.state_root)
+
+        if state_view.earliest_exit_epoch < activation_exit_epoch:
+            earliest_exit_epoch = activation_exit_epoch
+            exit_balance_to_consume = per_epoch_churn
+        else:
+            earliest_exit_epoch = state_view.earliest_exit_epoch
+            exit_balance_to_consume = state_view.exit_balance_to_consume
```

Comment:
```suggestion
        state = self.w3.cc.get_state_view(blockstamp.state_root)
        earliest_exit_epoch = max(state.earliest_exit_epoch, compute_activation_exit_epoch(blockstamp.ref_epoch)
        per_epoch_churn = get_activation_exit_churn_limit(self._get_total_active_balance(blockstamp))
        # New epoch for exits.
        if state.earliest_exit_epoch < earliest_exit_epoch:
            exit_balance_to_consume = per_epoch_churn
        else:
            exit_balance_to_consume = state.exit_balance_to_consume
```

just to be closer to the [source](https://github.com/ethereum/consensus-specs/blob/dev/specs/electra/beacon-chain.md#new-compute_exit_epoch_and_update_churn)

also would suggest to add docstring with smth like `inspired by <link_to_the_source>`

Reply:
I did write it in this way intentionally because readability is the fist priority. I can add a test to make sure we're compatible with the spec if you find it worth to do.

Reply:
I was just thinking about the rule that's always been around here, that we follow the spec literally, letter by letter,  wherever possible and left the referenced source link.
I don't have strong objections, but @F4ever might have some opinion here

Reply:
It also removes the need for the reader to validate that solution in part: you can just compare it to a reference that can be trusted